master/usage/migrating-pools.md

```diff
@@ -0,0 +1,183 @@
+
+- kube-apiserver's `--pod-network-cidr`
+- kube-proxy's `--cluster-cidr`
+- kube-controller-manager's `--cluster-cidr`
+
+OpenShift does not support changing the pod network CIDR (as per their [documentation on the `osm_cluster_network_cidr` configuration field](https://docs.openshift.org/latest/install_config/install/advanced_install.html#configuring-cluster-variables).
+
+**Application availability impact**
+
+This process will require the recreation of all {{ site.prodname }}-networked workloads, which will have some impact on
+the availability of your applications.
+
+## Consequences of deleting of an IP pool without following this migration procedure
+
+Removing an IP pool without following the migration procedure can cause network connectivity disruptions in any running
+workloads with addresses from that IP pool. Namely:
+
+- If IP-in-IP was enabled on the IP pool, those workloads will no longer have their traffic encapsulated.
+- If nat-outgoing was enabled on the IP pool, those workloads will no longer have their traffic NAT'd.
+- If using Calico BGP routing, routes to pods will no longer be aggregated.
+
+## Changing an IP pool
+
+The basic process is as follows:
+
+1. Add a new IP pool.
+2. Disable the old IP pool. This prevents new IPAM allocations from the old IP pool without affecting the networking of existing workloads.
+3. Recreate all existing workloads that were assigned an address from the old IP pool.
+4. Remove the old IP pool.
+
+### Example: Kubernetes
+
+In this example, we created a cluster with Kubeadm.  We wanted the pods to use IPs in the range
+`10.0.0.0/16` so we set `--pod-network-cidr=10.0.0.0/16` when running `kubeadm init`.  However, we
+installed {{ site.prodname }} without setting  default IP pool to match. Running `calicoctl get ippool -o wide` shows
+{{site.prodname}} created its default IP pool of `192.168.0.0/16`:
+
+```
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     false
+```
+
+Based on the output of `calicoctl get wep --all-namespaces`, we see `kube-dns` has already been allocated an address
+from the wrong range:
+
+```
+NAMESPACE     WORKLOAD                   NODE      NETWORKS            INTERFACE
+kube-system   kube-dns-6f4fd4bdf-8q7zp   vagrant   192.168.52.130/32   cali800a63073ed
+```
+
+Let's get started.
+
+1. Add a new IP pool:
+
+   ```
+   calicoctl create -f -<<EOF
+   apiVersion: projectcalico.org/v3
+   kind: IPPool
+   metadata:
+     name: new-pool
+   spec:
+     cidr: 10.0.0.0/16
+     ipipMode: Always
+     natOutgoing: true
+   EOF
+   ```
+
+   We should now have two enabled IP pools, which we can see when running `calicoctl get ippool -o wide`:
+
+   ```
+   NAME                  CIDR             NAT    IPIPMODE   DISABLED
+   default-ipv4-ippool   192.168.0.0/16   true   Always     false
+   new-pool              10.0.0.0/16      true   Always     false
+   ```
+
+2. Disable the old IP pool.
```

Comment:
Procedure reads really well. I would suggest that we could potentially lose the nested list though. That could easily be accomplished by deleting line 102 and would make the procedure even easier to follow.